cli/crates/server/src/proxy.rs

```diff
@@ -84,12 +88,14 @@ async fn start_inner(listener: TcpListener) -> Result<(), ServerError> {
             client,
         });
 
-    axum::Server::from_tcp(listener)
-        .map_err(ServerError::StartProxyServer)?
-        .http1_preserve_header_case(true)
-        .serve(router.into_make_service())
-        .await
-        .map_err(ServerError::StartProxyServer)?;
+    axum::serve(
+        tokio::net::TcpListener::from_std(listener).map_err(ServerError::StartProxyServer)?,
+        router,
+    )
+    // FIXME: Bring back!
```

Comment:
Might be breaking, we had various tests that expected cased headers (which is why I preserved the behavior here). Not sure it's critical though

Reply:
Yes, this is definitely something to address. Sadly, the new `axum::serve()` interface doesn't allow any configuration, and opting out of it would require a lot of boilerplate. I'm thinking, however, of doing something like [this](https://github.com/grafbase/grafbase/commit/a8c785a592cc03caf951a38f7a640111842ee17b). This is based on a [proposed change](https://github.com/grafbase/axum/commit/f9f1e4299ce247604ef2975171d0662e0fff5008) to `axum` from our fork. Do you think it makes sense? If it does, I'd try to send it upstream.

Reply:
It'd work, but perhaps 
```rs
    axum::build().preserve_header_case(true).serve(
         tokio::net::TcpListener::from_std(listener).map_err(ServerError::StartProxyServer)?,
         router
     )
```
?

Reply:
(or `Builder` or something similar)

Reply:
That configuration option is specific to HTTP1, though, hence why it's under `http1()`. What they're saying in the documentation for `serve()` is that if one wants to customise the server's configuration, one ought to use hyper's `Server` directly, which I am not a fan of. Hmm…

Reply:
@yoav-lavi what do you think we should do in the mean time? I feel like proceeding with the fork is odd since they're unlikely to accept the change. And if they could accept it, I wouldn't want to block on that happening. Is this setting critical enough?

BTW, you said:
> we had various tests that expected cased headers
To be fair, all the tests passed before I implemented this new way to turn that setting on, so I wonder where it would be critical?

Reply:
Actually, this PR also doesn't pull in that fork (I did it in a separate branch), and it does pass.

Reply:
@jakubadamw I think we disabled those once we removed the DB (they were mostly auth related), but I was thinking more of client implementations that don't take into account that headers can be lowercase and may break. We might be able to do that and let them know though.

Reply:
Since for now on prem isn't used yet we probably can proceed without the fork but we may want to check